feature-libs/cart/base/core/utils/utils.ts

```diff
@@ -38,6 +38,18 @@ export function isCartNotFoundError(error: ErrorModel): boolean {
   );
 }
 
+export function voucherExceededError(error: ErrorModel): boolean {
+  return error.message === 'coupon.already.redeemed';
+}
+
+export function voucherInvalidError(error: ErrorModel): boolean {
+  return error.message === 'coupon.invalid.code.provided';
+}
+
+export function isVoucherError(error: ErrorModel): boolean {
+  return error.type === 'VoucherOperationError';
+}
+
```

Comment:
Can you add some unit tests for them, otherwise our test coverage will be lower.

Reply:
Yes! will do